src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -124,7 +126,7 @@ public class TopicPartitionChannel {
    * <p>This boolean is used to indicate that we reset offset in kafka and we will only buffer once
    * we see the offset which is one more than an offset present in Snowflake.
    */
-  private boolean isOffsetResetInKafka;
+  private boolean isOffsetResetInKafka = false; // TODO @rcheng question: atomic?
```

Comment:
The offsets (offsetPersistedInSnowflake, processedOffset, latestConsumerOffset) are AtomicLongs implying that there is some parallelism considerations. Should this isOffsetResetInKafka be an AtomicBoolean as well? I think it may be overkill, but all other variables are Atomic, final or uses a lock (StreamingBuffer)

Reply:
I don't think we run anything in parallel, do we?

Reply:
We shouldn't and I can't think of a scenario where it would apply so I removed it. Just was wondering because we have other parallelization considerations